### ISSUE TITLE
Nexe is ignoring system truststore

### ISSUE THREAD

**Issue by deluxor** [nexe/nexe#721]
**What happened**:
When you call a URL with a self-signed certificate and you have your root-ca added in the system truststore nexe is ignoring it.
 
**What you expected to happen**:
When an application is bundled with nexe it should be able to use the root-ca certificates added in the system truststore, so it can proceed with the https handshake. For instance when running the script directly with system's node runtime: `node index.js` it's ok.

**How to reproduce it (as minimally and precisely as possible)**:
```
const https = require('https')
const options = {
  hostname: 'whatever.com',
  port: 443,
  path: '/todos',
  method: 'GET'
}

const req = https.request(options, res => {
  console.log(`statusCode: ${res.statusCode}`)

  res.on('data', d => {
    process.stdout.write(d)
  })
})

req.on('error', error => {
  console.error(error)
})

req.end()
```
Build the bundle:
```root@container:/# nexe -i index.js -o x
ℹ nexe 3.3.2
✔ Downloading...100%
✔ Compiling result
✔ Entry: 'index.js' written to: x
✔ Finished in 6.965s
root@container:/# ./x
{ Error: unable to get local issuer certificate
    at TLSSocket.onConnectSecure (_tls_wrap.js:1051:34)
    at TLSSocket.emit (events.js:189:13)
    at TLSSocket._finishInit (_tls_wrap.js:633:8) code: 'UNABLE_TO_GET_ISSUER_CERT_LOCALLY' }
root@container:/#
```
**Environment**:
- Platform(OS/Version): Linux/x64
- Host Node Version: 12
- Target Node Version: 12
- Nexe version: 3.3.2
- Python Version: 2.7.16

**Comment by jaredallard** [comment]
I'm not sure this is specific to `nexe`. When I attempt to use self-signed certificates with node on OSX I run into this same issue all the time (I use Debookee to MTIM SSL connections for reverse engineering)

**Comment by deluxor** [comment]
> I'm not sure this is specific to `nexe`. When I attempt to use self-signed certificates with node on OSX I run into this same issue all the time (I use Debookee to MTIM SSL connections for reverse engineering)
> 
> But yes, this does warrant investigation.

Yes, self-signed certificates sometimes are a serious problem to detect, however if I run the script with the system's node runtime it's ok (with the root-ca added to the system  truststore).

**Comment by daguej** [comment]
node has sadly _never_ supported the use of the system trust store on Windows and Mac, and the situation on Linux is murky.

Historically, node bundled a hard-coded, built-in list of trusted CAs (copied from Mozilla).  The only option to override was in JS, where you can replace this trust store completely by passing the [`ca` option to `tls.createSecureContext`](https://nodejs.org/dist/latest-v13.x/docs/api/tls.html#tls_tls_createsecurecontext_options) (or an API that wraps it, like `https.request`).

node still uses this approach today, but there have been a few configuration hooks added:

- Set the [`NODE_EXTRA_CA_CERTS`](https://nodejs.org/api/cli.html#cli_node_extra_ca_certs_file) env var at runtime to specify additional CA cert(s)
- Pass [`--use-openssl-ca`](https://nodejs.org/api/cli.html#cli_use_bundled_ca_use_openssl_ca) at runtime to use the OpenSSL CA store (which is the system store only on Linux)
- Configure `--openssl-use-def-ca-store` at *build* time so that node defaults to the OpenSSL CA store instead of the bundled CA store.

The official build of node distributed by nodejs.org uses the bundled store by default on all platforms, while builds of node distributed by many Linux distros enable `--openssl-use-def-ca-store`.  My guess is that you're using a build of node with this option enabled.

So in short, the default and unfortunate expected behavior is that node does **not** use your system CA store.  I would expect nexe to behave the same.

There are modules on npm that add support for each platform's CA trust store (Windows certmgr/Mac Keychain, Linux OpenSSL store) independently of node's configuration; I'd use one of them.
